services/wallet/transfer/commands_sequential.go

```diff
@@ -241,7 +241,18 @@ func (c *findNewBlocksCommand) Run(parent context.Context) error {
 	}
 
 	if len(accountsWithDetectedChanges) != 0 || c.iteration%c.logsCheckIntervalIterations == 0 {
-		err = c.findAndSaveTokenBlocks(parent, c.fromBlockNumber, headNum)
+		from := big.NewInt(0).Set(c.fromBlockNumber)
+		for _, account := range c.accounts {
+			blockRange, _, err := c.blockRangeDAO.getBlockRange(c.chainClient.NetworkID(), account)
+			if err != nil {
+				return err
+			}
+
+			if blockRange.tokens.LastKnown != nil || blockRange.tokens.LastKnown.Cmp(zero) != 0 && blockRange.tokens.LastKnown.Cmp(from) == -1 {
+				from = big.NewInt(0).Set(blockRange.tokens.LastKnown)
```

Comment:
No it is not and it shouldn't be covered by `findBlocksCommand`. The issue is like every 5th iteration of `findNewBlocksCommand.Run` runs erc20/1155/721 check, but on each previous eth/erc20 check (which happens every 2 mins ) `c.fromBlockNumber` is updated, that makes it irrelevant to erc20/1155/721 check.

Reply:
By the way I planned to replace `findBlocksCommand ` with `findNewBlocksCommand` right after login, that's a way to save some requests  (specifically eth_getBalance and eth_getHeader). So that `findBlocksCommand` will be used only for initial scan when needed, and when we will implement partial loading of history it will also be used for loading older parts of history.